static/js/formatters-internal.js

```diff
@@ -311,164 +311,117 @@ export function image(simpleOrComplexImage = {}, desiredSize = '200x', atLeastAs
-  const [urlBeforeDimensions, dimensions] = _splitStringOnIndex(urlWithoutExtension, urlWithoutExtension.lastIndexOf('/') + 1);
-  const fullSizeDims = dimensions.split('x');
-
-  let desiredWidth, desiredHeight;
+function _getImageFormatOptions(desiredSize, atLeastAsLarge, fullSizeWidth, fullSizeHeight) {
   let desiredDims = desiredSize.split('x');
+  const desiredWidthProvided = desiredDims[0] !== '' && desiredDims[0] !== '1';
+  const desiredHeightProvided = desiredDims[1] !== '' && desiredDims[1] !== '1';
 
-  if (desiredDims[0] !== '') {
+  // both dimensions are not provided, return original image
+  if (!desiredWidthProvided && !desiredHeightProvided) {
+    return '';
+  }
+
+  const originalRatio =
+    (!!fullSizeWidth && !!fullSizeHeight) ? (fullSizeWidth / fullSizeHeight) : undefined;
+  let desiredWidth;
+  let desiredHeight;
+  let formatOptions = ['fit=contain'];
+  if (desiredWidthProvided) {
     desiredWidth = Number.parseInt(desiredDims[0]);
     if (Number.isNaN(desiredWidth)) {
       throw new Error("Invalid width specified");
     }
-  } else {
-    desiredWidth = atLeastAsLarge ? 1 : Number.parseInt(fullSizeDims[0]);
   }
-
-  if (desiredDims[1] !== '') {
+  if (desiredHeightProvided) {
     desiredHeight = Number.parseInt(desiredDims[1]);
     if (Number.isNaN(desiredHeight)) {
       throw new Error("Invalid height specified");
     }
-  } else {
-    desiredHeight = atLeastAsLarge ? 1 : Number.parseInt(fullSizeDims[1]);
   }
 
-  const urlWithDesiredDims = urlBeforeDimensions + desiredWidth + 'x' + desiredHeight + extension;
-
-  return atLeastAsLarge
-    ? _replaceUrlHost(urlWithDesiredDims, 'dynl.mktgcdn.com')
-    : _replaceUrlHost(urlWithDesiredDims, 'dynm.mktgcdn.com');
-}
+  // only width is provided
+  if (desiredWidthProvided && !desiredHeightProvided) {
+    formatOptions.push(`width=${desiredWidth}`);
 
-/**
- * Given an EU image url, returns the dynamic url.
- *
- * @param {Object} image The image object. (e.g.
- * {
- *   url: 'https://a.eu.mktgcdn.com/f/0/FLVfkpR1IwpWrWDuyNYCJWVYIDfPO6x1QSztXozMIzo.jpg',
- *   sourceUrl: 'https://a.mktgcdn.com/p/UN9RPhz0V9D8bNZ3XfNpkGnAk6ikFhVmgvntlBjVyMA/1200x675.jpg',
- *   width: 1200,
- *   height: 675,
- * })
- * @param {string} desiredSize The desired size of the image ('<width>x<height>')
- * @param {boolean} atLeastAsLarge Whether the image should be at least as large as the desired
- *                                 size in one dimension or smaller than the desired size in both
- *                                 dimensions.
- * @returns {string} A dynamic url (e.g. 'https://dyn.eu.mktgcdn.com/f/0/FLVfkpR1IwpWrWDuyNYCJWVYIDfPO6x1QSztXozMIzo.jpg/width=200,fit=contain')
- */
-function _getEuImageDynamicUrl(image, desiredSize, atLeastAsLarge) {
-  let fullSizeWidth, fullSizeHeight;
-  if (image.width) {
-    fullSizeWidth = image.width;
-  }
-  if (image.height) {
-    fullSizeHeight = image.height;
+    return `/${formatOptions.join(',')}`;
```

Comment:
can we perform this step once at the end before returning? it seems like it's repeated for each case

Reply:
I wanted to break early and avoid nested is/else. Ack